libs/angular-utils/src/lib/services/custom-use-style.service.ts

```diff
@@ -3,59 +3,109 @@ import { UseStyle } from 'primeng/usestyle'
 import { REMOTE_COMPONENT_CONFIG, RemoteComponentConfig } from '@onecx/angular-remote-components'
 import { AppStateService } from '@onecx/angular-integration-interface'
 import { firstValueFrom, map, ReplaySubject } from 'rxjs'
+import { THEME_OVERRIDES, ThemeOverrides } from '../theme/application-config'
+import { toVariables } from '@primeuix/styled'
 
 export const SKIP_STYLE_SCOPING = new InjectionToken<boolean>('SKIP_STYLE_SCOPING')
+const notCharacterOrDashRegex = /[^a-zA-Z0-9-]/g
 
 @Injectable({ providedIn: 'any' })
 export class CustomUseStyle extends UseStyle {
   constructor(
     private appStateService: AppStateService,
     @Optional() @Inject(SKIP_STYLE_SCOPING) private skipStyleScoping?: boolean,
-    @Optional() @Inject(REMOTE_COMPONENT_CONFIG) private remoteComponentConfig?: ReplaySubject<RemoteComponentConfig>
+    @Optional() @Inject(REMOTE_COMPONENT_CONFIG) private remoteComponentConfig?: ReplaySubject<RemoteComponentConfig>,
+    @Optional() @Inject(THEME_OVERRIDES) private themeOverrides?: ThemeOverrides
   ) {
     super()
   }
+  // PrimeNg defines CSS variables and styles globally in <style> elements
+  // Each Application needs to isolate the CSS variables and styles from others
   override use(css: any, options?: any): { id: any; name: any; el: any; css: any } {
-    this.getStyleIdentifier().then((scopedStyleId) => {
-      if (scopedStyleId !== '' && !(options.name as string).endsWith('-variables')) {
-        css = `
-        @scope([data-style-id="${scopedStyleId}"][data-no-portal-layout-styles]) to ([data-style-isolation]) {
-                ${css}
-            }
-        `
-      }
-      if (scopedStyleId === '' && !(options.name as string).endsWith('-variables')) {
-        css = `
-        @scope([data-style-id="shell-ui"]) to ([data-style-isolation]) {
-                ${css}
-            }
-        `
-      }
+    this.getScopeIdentifier().then((scopeId) => {
+      css = this.replacePrefix(css, scopeId)
+      css = this.isStyle(options.name as string) ? this.scopeStyle(css, scopeId) : css
+
       options = {
         ...options,
-        name: (options.name ?? '') + (scopedStyleId === '' ? scopedStyleId : '-' + scopedStyleId),
+        name: (options.name ?? '') + (scopeId === '' ? scopeId : '-' + scopeId),
       }
       super.use(css, options)
+      return this.applyOverrides(scopeId)
     })
+    // Fake response is returned because async action is done in sync context
+    // Fake response should indicate usage by displaying errors when dummy data is used
```

Comment:
This works because the result is not being used at the moment. The fake response with the errors ensures that we can detect future usages of the result.